webapp/src/widgets/VoiceActivityDetection/micAmplidute.ts

```diff
@@ -0,0 +1,55 @@
+import {StackLimited} from "../../utils/stack-limited";
+
+interface Options {
+  stream: MediaStream
+  intervalMs?: number
+  durationMs?: number
+  threshold?: number
+  onUpdate?: (value: number) => void
+  onDetect: (flag: boolean) => void
+}
+
+export function watchMicAmplitude(options: Options): () => void {
+  const { stream, intervalMs = 50, durationMs = 500, threshold = 1, onDetect } = options;
+
+  const audioContext = new window.AudioContext();
+  const analyser = audioContext.createAnalyser();
+  const dataArray = new Uint8Array(analyser.frequencyBinCount);
+  analyser.fftSize = 256;
+
+  const source = audioContext.createMediaStreamSource(stream);
+  source.connect(analyser);
+
+  const stackLimit = Math.round(durationMs / intervalMs)
+  const micData = new StackLimited(stackLimit)
+
+  let isThresholdReached = false;
+
+  function measureMicrophoneLevel(): ReturnType<typeof setTimeout> {
+    analyser.getByteFrequencyData(dataArray);
```

Comment:
Since we have access to frequency sample data, should we look at only typical ranges of voice frequency, so the mic is not activated by noise or other sounds? https://en.wikipedia.org/wiki/Voice_frequency